airbyte-webapp-e2e-tests/cypress/integration/connection.spec.ts

```diff
@@ -62,41 +66,125 @@ describe("Connection main actions", () => {
     deleteDestination(destName);
   });
 
+  it("Connection sync mode Incremental Append", () => {
+    const sourceName = appendRandomString("Test connection Postgres source cypress");
+    const destName = appendRandomString("Test connection Postgres destination cypress");
+
+    cy.intercept("/api/v1/web_backend/connections/update").as("updateConnection");
+
+    createTestConnection(sourceName, destName);
+
+    goToSourcePage();
+    openSourceDestinationFromGrid(sourceName);
+    openSourceDestinationFromGrid(destName);
+
+    goToReplicationTab();
+
+    searchStream("admins");
+    selectSyncMode("Incremental", "Append");
+    selectCursorField("email");
+
+    submitButtonClick();
+    confirmStreamConfigurationChangedPopup();
+
+    cy.wait(5000);
+    cy.wait("@updateConnection").then((interception) => {
+      assert.isNotNull(interception.response?.statusCode, "200");
+    });
+
+    checkSuccessResult();
+
+    goToSourcePage();
+    openSourceDestinationFromGrid(sourceName);
+    openSourceDestinationFromGrid(destName);
+
+    goToReplicationTab();
+
+    searchStream("admins");
+
+    checkCursorField("email");
```

Comment:
Given that the test is named "Connection sync mode Incremental Append", this code should check the sync mode field and not the cursor field. In fact, as written there is no difference between the assertions in this test and in the following one: in fact, if there were a backend bug which saved every connection with `Full refresh | Append` sync mode (exactly the kind of error this test should fail on!), both tests would still pass.

Reply:
Actually we cannot same Full refresh | Append sync mode without cursor field so it looks like as a necessary details.